spec/Section 4 -- Introspection.md

```diff
@@ -500,3 +510,76 @@ Fields\:
+directly expressible in the type system. This may include experimental GraphQL
+features (such as new syntax or behavior), protocol support (such as GraphQL
+over WebSockets), or additional operational metadata (such as release
+identifiers). Capabilities may be supplied by the GraphQL implementation, the
+service, or both.
+
+The `__Service.capabilities` field exposes a _capability_ list. A _capability_
+consists of a capability identifier and optionally a string value.
+
+**Capability identifier**
+
+A capability identifier is a string value composed of two or more segments
+separated by a period (`.`). Each segment must begin with an ASCII letter, and
+must contain only ASCII letters, digits and hyphens. These constraints are
+inspired by reverse domain name notation to encourage global uniqueness and
+collision-resistance. Unlike the domain name system, capability identifiers are
+case sensitive. Identifiers beginning with the prefix {"org.graphql."} are
+reserved and must not be used outside of official GraphQL Foundation
+specifications. Further, identifiers beginning with the prefix
+{"org.graphql.http."} are reserved for use by the GraphQL-over-HTTP
+specification, and identifiers beginning with the prefix {"org.graphql.rfc."}
+are reserved for RFC proposals.
+
+Identifiers defined by specific projects, vendors, or implementations should
+begin with a prefix derived from a DNS name they control (e.g.,
+{"com.example."}).
+
+Clients should compare identifiers using exact string equality and should ignore
+unknown identifiers.
+
+Implementers should not change the meaning of capability identifiers; instead, a
+new capability identifier should be used when the meaning changes. Implementers
+should ensure that capability identifiers remain stable and version-agnostic
+where possible; capability versioning, if needed, can be indicated using dot
+suffixes (e.g.{ "org.example.capability.v2"}).
+
+This system enables incremental feature adoption and richer tooling
+interoperability, while avoiding tight coupling to specific implementations.
+
+**Capability value**
+
+The value assigned to a given capability may either be {null} to simply indicate
+the capability is supported, or a string to provide additional information. For
+example {"org.graphql.onError"} does not require additional information and thus
+uses the value {null}, whereas {"org.graphql.defaultErrorBehavior"} needs the
+{value} to indicate which _error behavior_ is the default.
+
+**Specified capabilities**
+
+This version of the specification defines the following capabilities:
+
+- {"org.graphql.defaultErrorBehavior"} - indicates the _default error behavior_
+  of the service via the {value}. If not present, must be assumed to be
+  {"PROPAGATE"}.
+- {"org.graphql.onError"} - indicates that the service allows the client to
+  specify {onError} in a request to indicate the _error behavior_ the service
+  should use for the request. {value} is {null}.
```

Comment:
What about this?

```suggestion
- {"org.graphql.defaultErrorBehavior"} - indicates the _default error behavior_
  of the service via the {value}. 
  If present, the service must be assumed to support the {onError} parameter 
  to indicate the _error behavior_ on requests.
  If not present, must be assumed to be {"PROPAGATE"}.
```

Less invalid cases possible.

Reply:
This requires services to support the `onError` request parameter forever, I don't think we should do that. If a service wants to start up that only supports the `NO_PROPAGATE` error behavior and doesn't want to allow clients to influence that, I think it should be allowed to.

Reply:
SGTM 👍